### PR TITLE
7.0 - l10n_canada rename column in module res_partner_attributes_add_SIN

### PR DESCRIPTION
**Pull request by ghost**
nas is a french term, meaning numéro d'assurance social. I change it to the english term sin, social insurance number.

### CODE REVIEW

**Comment by max3903** [comment]
:+1:

**Comment by bwrsandman** [comment]
The migration script won't run until you set the version to 1.1

**Comment by bwrsandman** [comment]
:-1: 

You should bump the version of the module to 1.2 if you're changing the name of a field. Adjust your migration script accordingly.

File: res_partner_attributes_add_SIN/migrations/7.0.1.1/post-migration.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+
+def migrate(cr, version):
+    if version and version < '7.0.1.1':
```
Comment (bwrsandman) [inline res_partner_attributes_add_SIN/migrations/7.0.1.1/post-migration.py]:
``` python
    if not version:
        return
```

The version is taken from the folder.

File: res_partner_attributes_add_SIN/migrations/7.0.1.1/post-migration.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        cr.execute("ALTER TABLE res_partner RENAME COLUMN nas TO sin;")
```
Comment (bwrsandman) [inline res_partner_attributes_add_SIN/migrations/7.0.1.1/post-migration.py]: This is incomplete, please refer to the [OpenUpgrade function](https://github.com/OpenUpgrade/OpenUpgrade/blob/8.0/openerp/openupgrade/openupgrade.py#L138) and copy it.

File: res_partner_attributes_add_SIN/migrations/7.0.1.1/post-migration.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
```
Comment (bwrsandman) [inline res_partner_attributes_add_SIN/migrations/7.0.1.1/post-migration.py]: Column renames should be done in pre-migration scripts.

Rename to pre-migration.py

https://doc.therp.nl/openupgrade/API.html#openupgrade.rename_columns

File: res_partner_attributes_add_SIN/migrations/7.0.1.2/pre-migration.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+import logging
+logger = logging.getLogger('OpenUpgrade')
+
+
+def migrate(cr, version):
+    if not version:
+        return
+
+    if version < '7.0.1.2':
```
Comment (bwrsandman) [inline res_partner_attributes_add_SIN/migrations/7.0.1.2/pre-migration.py]: This condition is not necessary.

Please factor the alter table and drop index into a function with a doctring.

You can just copy the OpenUpgrade function as is. It is cleaner that way.

**Comment by ghost** [comment]
@bwrsandman When I test, the migration script is not called, any idea why?

**Comment by bwrsandman** [comment]
@dufresnedavid are you testing with res_partner_attributes_add_SIN installed 1.0, then updating the code to 1.2, then updating the module?

**Comment by ghost** [comment]
@bwrsandman Exactly, that's what I'm doing.

**Comment by veloutin** [comment]
:+1:

**Comment by llacroix** [comment]
:+1:

**Comment by bwrsandman** [comment]
@dufresnedavid est-ce que le script de migration marche finalement?

**Comment by bwrsandman** [comment]
:+1:

**Pull request merged**